sstables/sstables.cc

```diff
@@ -1250,6 +1250,25 @@ sstable::find_first_position_in_partition(reader_permit permit, const dht::decor
     co_return std::move(ret);
 }
 
+future<> sstable::load_first_and_last_position_in_partition() {
+    if (!_schema->clustering_key_size()) {
+        co_return;
+    }
+    auto& sem = _manager.sstable_metadata_concurrency_sem();
+    reader_permit permit = co_await sem.obtain_permit(&*_schema, "sstable::load_first_and_last_position_range", sstable_buffer_size, db::no_timeout);
```

Comment:
Shouldn't we use a timeout here? We can use a generous one, it will still allow us to avoid a complete silent lock-up.

Reply:
No, our internal queries are usually without a timeout. We did regret using infinite timeouts in some cases though.